### PR TITLE
EMA isolation on 5L/256d + T_max=50 (clean single-variable test)

### PR DESCRIPTION
**Pull request by morganmcg1**
# Hypothesis: EMA + 5L/256d + T_max=50 (Clean Isolation)

**Wave 13 assignment for haku**

## Hypothesis

EMA (Exponential Moving Average) of model weights has not been cleanly tested on the current best recipe (5L/256d + T_max=50). Prior EMA tests used the old 4L/256d baseline (nezuko #383). This experiment isolates the EMA effect on top of the current strongest configuration.

EMA smooths the parameter trajectory and typically provides free ~0.2–0.5% improvement on validation metrics by averaging out the noise in the final training phase. With the T_max=50 cosine schedule providing a slow final decay, EMA at decay=0.9995 should integrate well.

## Baseline

| Metric | Current Best (val) | AB-UPT Target |
|--------|-------------------|--------------|
| `val_primary/abupt_axis_mean_rel_l2_pct` | **6.9549** | 4.51 |
| `val_primary/surface_pressure_rel_l2_pct` | 4.5644 | 3.82 |
| `val_primary/volume_pressure_rel_l2_pct` | 3.9361 ✓ | 6.08 |
| `val_primary/wall_shear_y_rel_l2_pct` | 8.7345 | 3.65 |
| `val_primary/wall_shear_z_rel_l2_pct` | 10.5766 | 3.63 |

Best checkpoint: ep~45.3, step 807,025, W&B run `vu4jsiic` (PR #174).

**Beat this**: `val_primary/abupt_axis_mean_rel_l2_pct` < **6.9549%**

## Architecture

- **Layers**: 5 (identical to current best)
- **Hidden dim**: 256 (identical to current best)
- **Heads**: 4 (identical to current best)
- **PE**: FourierEmbed (`--fourier-pe`, mandatory)
- **LR schedule**: Cosine with T_max=50
- **EMA**: **ENABLED** — `--use-ema --ema-decay 0.9995` (the single change vs baseline)

This is a clean single-variable test: EMA on vs EMA off, holding everything else at the current best recipe.

## Training Instructions

Run for 50 epochs on 4×H100 GPUs:

```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --fourier-pe \
  --use-ema \
  --ema-decay 0.9995 \
  --lr 3e-4 \
  --lr-cosine-t-max 50 \
  --no-compile-model \
  --wandb-group bengio-wave13
```

## Gate Schedule

Report `val_primary/abupt_axis_mean_rel_l2_pct` at each gate. Kill and report if metric is NOT below the gate threshold.

| Epoch | Gate threshold | Action if NOT below |
|-------|---------------|---------------------|
| ep5   | < 13.0%       | Kill + report       |
| ep10  | < 10.0%       | Kill + report       |
| ep15  | < 8.9%        | Kill + report       |
| ep30  | < 6.9549%     | Kill + report       |
| ep50  | Final         | Report best checkpoint |

## ACK Required

Reply to this PR with "ACK" within 30 minutes of assignment. Failure to ACK results in immediate PR closure.

## Notes

- `--no-compile-model` is mandatory (PyTorch 2.x Inductor crash at validation)
- `--fourier-pe` is mandatory for comparability (n_params = 3,249,813 for 5L/256d — report actual n_params to confirm)
- Report W&B run ID immediately after launching
- Do NOT run unauthorized side experiments
- EMA validation metrics should be lower than non-EMA metrics in late training — report both EMA and non-EMA val metrics if your training script logs both

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: URGENT — ACK OVERDUE. PR #438 was assigned at 17:52Z; it is now past the 30-minute ACK deadline. Post your W&B run ID and confirmation that training has started IMMEDIATELY. Failure to ACK in the next 15 minutes will result in PR closure and compliance action.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #438 because COMPLIANCE: PR #438 closed. ACK deadline was 18:22Z (30 minutes after assignment at 17:52Z). It is now 18:40Z — 48 minutes past deadline with no ACK and no W&B run ID posted. This is consistent with the pattern that led to Wave 10 compliance actions. haku is now IDLE. New assignment will follow promptly — ensure you ACK within 30 minutes of the new PR or GPU access will be suspended..
